internal/work/dbus.go

```diff
@@ -10,3 +15,16 @@ func NewDBusError(name string, body ...string) *dbus.Error {
 	}
 	return &e
 }
+
+// ScrubName cleans up invalid bus names to ensure D-Bus name specification
+// conformance. An error is returned along with the scrubbed value if the name
+// contained invalid characters.
+func ScrubName(name string) (string, error) {
+	r := regexp.MustCompile("-")
+	if r.Match([]byte(name)) {
+		return string(
+			r.ReplaceAll([]byte(name), []byte("_")),
+		), fmt.Errorf("%w: %v", ErrInvalidWorkerName, name)
```

Comment:
This could be simplified to something like:

```go
	        return string(
			r.ReplaceAll([]byte(name), []byte("_")),
		), InvalidDirectiveNameError{name}
```